modules/navier_stokes/include/userobjects/INSFVRhieChowInterpolatorSegregated.h

```diff
@@ -0,0 +1,124 @@
+#include "CellCenteredMapFunctor.h"
+#include "FaceCenteredMapFunctor.h"
+#include "VectorComponentFunctor.h"
+#include <unordered_map>
+#include <set>
+#include <unordered_set>
+
+class MooseMesh;
+class INSFVVelocityVariable;
+class INSFVPressureVariable;
+namespace libMesh
+{
+class Elem;
+class MeshBase;
+}
+
+/**
+ * A user object which implements the Rhie Chow interpolation for segregated
+ * momentum-pressure systems.
+ */
+class INSFVRhieChowInterpolatorSegregated : public RhieChowInterpolatorBase
+{
+public:
+  static InputParameters validParams();
+  INSFVRhieChowInterpolatorSegregated(const InputParameters & params);
+
+  /// Get the face velocity (used in advection terms)
+  VectorValue<ADReal> getVelocity(const Moose::FV::InterpMethod m,
+                                  const FaceInfo & fi,
+                                  const Moose::StateArg & time,
+                                  const THREAD_ID tid) const override;
+
+  /// Initialize the container for face velocities
+  void initFaceVelocities();
+  /// Update the values of the face velocities in the containers
+  void computeFaceVelocity();
+  /// Update the cell values of the velocity variables
+  void computeCellVelocity();
+
+  /// We disable this for the segregated solver
+  void addToA(const libMesh::Elem * /*elem*/,
+              unsigned int /*component*/,
+              const ADReal & /*value*/) override
+  {
+    mooseError(
+        "addToA function is not implemented for the RhieChow interpolation in segregated solvers.");
+  }
+
+  void meshChanged() override;
+  void initialize() override;
+  void execute() override {}
+  void finalize() override {}
+
+  bool segregated() const override { return true; };
+
+  /// Update the momentum system-related information
+  void linkMomentumSystem(std::vector<NonlinearSystemBase *> momentum_systems,
+                          const std::vector<unsigned int> & momentum_system_numbers,
+                          const TagID & pressure_gradient_tag);
+
+  /**
+   * Computes the inverse of the digaonal (1/A) of the system matrix plus the H/A components for the
+   * pressure equation plus Rhie-Chow interpolation.
+   */
+  void computeHbyA(bool verbose);
+
+protected:
+  /// Populatethe face values of the H/A
+  void populateHbyA(const std::vector<std::unique_ptr<NumericVector<Number>>> & raw_hbya,
+                    const std::vector<unsigned int> & var_nums);
+  /**
+   * A map functor from faces to $HbyA_{ij} = (A_{offdiag}*\mathrm{(predicted~velocity)} -
+   * \mathrm{Source})_{ij}/A_{ij}$. So this contains the offdiagonal part of the system matrix
+   * multiplied by the predicted velocity minus the source terms from the right hand side of the
+   * linearized momentum predictor step.
+   */
+  FaceCenteredMapFunctor<RealVectorValue, std::unordered_map<dof_id_type, RealVectorValue>> _HbyA;
+
+  /**
+   * We hold on to the raw values of the HbyA vectors so that we can easily reconstruct the
```

Comment:
What makes it raw? When I see "raw values", I think of `MetaPhysicL::raw_value`, and I don't think that's the right concept here

Reply:
I rewrote this, by raw I mean they are just petsc vectors so they don't have functoriality in contrast with the CCMF.